### ISSUE TITLE
Latest @types/reach_router release makes GatsbyLinkProps innerRef type incompatible

### ISSUE THREAD

**Issue by thiagoods** [gatsbyjs/gatsby#22706]
## Description

With the release of a change to the type of the innerRef property inside LinkProps, 
the GatsbyLinkProps interface is wrong. That breaks any Gatsby build using Typescript due to incompatible types.

Context here:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43367

### InnerRef prop currently defined in LinkProps interface

```typescript
innerRef?: React.RefCallback<HTMLAnchorElement>;
```

### Current innerRef type in GatsbyLinkProps interface

```typescript
innerRef?: Function;
```

**Comment by pieh** [comment]
Ok, so likely fix here is to drop "our" `innerRef` and bump `@types/reach__router`?

Also how exactly does it break? I tried one of typescript starters and it didn't give me any problems (even when using `innerRef`)

**Comment by pieh** [comment]
This should be fixed in `gatsby@2.20.11` / `gatsby-link@2.3.2`
